From 8a9a4c75c0591a0fd805f6d18194d1e8fc06bd4a Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Tue, 21 Nov 2023 21:37:03 +0530
Subject: [PATCH 2/2] wifi: ath12k: initialize link_id and link status in
 ath12k_dp_mon_rx_deliver_msdu()

Initialize link_id and link status in ath12k_dp_mon_rx_deliver_msdu().
Uninitialized link_id and link status values can lead to undefined behaviour
in mac80211.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_mon.c b/drivers/net/wireless/ath/ath12k/dp_mon.c
index 1d33bcc..3e9ae7f 100644
--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -2173,6 +2173,9 @@ static void ath12k_dp_mon_rx_deliver_msdu(struct ath12k *ar, struct napi_struct
 		status->flag |= RX_FLAG_RADIOTAP_HE;
 	}
 
+	status->link_valid = 0;
+	status->link_id = 0;
+
 	if (!(status->flag & RX_FLAG_ONLY_MONITOR))
 		decap = ppduinfo->cmn_mpdu_info.mon_mpdu[mpdu_idx].msdu_info[0].decap_format;
 	spin_lock_bh(&ar->ab->base_lock);
-- 
2.34.1

